dockerfiles/init/modules/che-kubernetes-helm/templates/ingress.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: che-ingress
+  annotations:
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    kubernetes.io/ingress.class: "nginx"
+{{- if .Values.tlsEnabled }}
+    kubernetes.io/tls-acme: "true"
+{{- else }}
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+{{- end }}
+    ingress.kubernetes.io/proxy-read-timeout: "3600"
+    ingress.kubernetes.io/proxy-connect-timeout: "3600"
+spec:
+{{- if .Values.tlsEnabled }}
+  tls:
+  - hosts:
+    - {{ template "cheHost" . }}
+    secretName: che-tls
+{{- end }}
+  rules:
+  - http:
+      paths:
+      - backend:
+          serviceName: che-host
+          servicePort: 8080
+        path: /
+{{- if .Values.isHostBased }}
```

Comment:
I think that this `if` condition should be removed: it's already done in `cheHost` template and it's not consistent with how `CHE_HOST` and `CHE_API`are defined.

Reply:
We're working on improving host-based routing support in Kubernetes. We'll cleanup the ingress in the process.